File: backend-ms1/backend-ms1/src/main/java/ma/zs/emailling/service/facade/admin/emailgroup/GroupeEmailAdminService.java
package ma.zs.emailling.service.facade.admin.emailgroup;

import java.util.List;
import ma.zs.emailling.bean.core.emailgroup.GroupeEmail;
import ma.zs.emailling.dao.criteria.core.emailgroup.GroupeEmailCriteria;
import ma.zs.emailling.zynerator.service.IService;



public interface GroupeEmailAdminService extends  IService<GroupeEmail,GroupeEmailCriteria>  {

    List<GroupeEmail> findByAdminGroupeId(Long id);
    int deleteByAdminGroupeId(Long id);
    long countByAdminGroupeId(Long id);




}

File: frontend/src/app/shared/service/admin/emailgroup/GroupeEmailAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {GroupeEmailDto} from 'src/app/shared/model/emailgroup/GroupeEmail.model';
import {GroupeEmailCriteria} from 'src/app/shared/criteria/emailgroup/GroupeEmailCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class GroupeEmailAdminService extends AbstractService<GroupeEmailDto, GroupeEmailCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): GroupeEmailDto {
        return new GroupeEmailDto();
    }

    public constrcutCriteria(): GroupeEmailCriteria {
        return new GroupeEmailCriteria();
    }

    get API() {
        return environment.apiUrlEmaillingservice + 'admin/groupeEmail/';
    }
}
